packages/ui/src/SelectField.tsx

```diff
@@ -184,6 +242,35 @@ export const SelectField = <V,>({
     }
   }, [menuPortalTarget])
 
+  const onChangeMultipleFn = React.useCallback(
+    (values: SelectFieldOption<V>[] | null) => {
+      ;(onChange as (newValue: V[] | null) => void)(values === null ? null : values.map(({ value }) => value))
+    },
+    [onChange],
+  )
```

Comment:
Since empty `multi select` now returns an empty array when there's nothing selected, we might want to remove all these ugly `null`'s from other places:

```suggestion
  const onChangeMultipleFn = React.useCallback(
    (values: SelectFieldOption<V>[]) => {
      ;(onChange as (newValue: V[]) => void)(values.map(({ value }) => value))
    },
    [onChange],
  )
```

@jvorcak Can you also check other places too?